components/infobox/commons/infobox_weapon.lua

```diff
@@ -0,0 +1,108 @@
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Infobox/Weapon
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Namespace = require('Module:Namespace')
+local BasicInfobox = require('Module:Infobox/Basic')
+local Flags = require('Module:Flags')
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+local Header = Widgets.Header
+local Title = Widgets.Title
+local Center = Widgets.Center
+local Customizable = Widgets.Customizable
+
+local Weapon = Class.new(BasicInfobox)
+
+function Weapon.run(frame)
+	local weapon = Weapon(frame)
+	return weapon:createInfobox()
+end
+
+function Weapon:createInfobox()
+	local infobox = self.infobox
+	local args = self.args
+
+	local widgets = {
+	Header{
+		name = self:nameDisplay(args),
+		image = args.image,
```

Comment:
Customization point for the subheader maybe?

Reply:
the header widget supports subheader, could make use of that

Reply:
Is there a benefit to using a customizable subheader over just header?

Reply:
imo either make the whole header customizable or just use `subHeader = self:subHeader(args)` in the current header together with a function of the kind
```lua
function Weapon:subHeader(args)
	return nil
end
```

Reply:
Header is usually just the infobox/page name and the images used to show what the infobox is right? In this case it would be just the weapons image. Would we really want the header customizable if the page name is already the title of the infobox? I cant think of any cases where the header would be different

Reply:
The header text is already customizable (`name = self:nameDisplay(args)`)
I don't see a reason not to add the option to add a subheader without duplicating all the image stuff.